dd-trace-core/src/main/java/datadog/trace/core/CoreTracer.java

```diff
@@ -559,10 +559,13 @@ private CoreTracer(
     this.initialSampler = sampler;
 
     // Get initial Trace Sampling Rules from config
-    TraceSamplingRules traceSamplingRules =
-        config.getTraceSamplingRules() == null
-            ? TraceSamplingRules.EMPTY
-            : TraceSamplingRules.deserialize(config.getTraceSamplingRules());
+    String traceSamplingRulesJson = config.getTraceSamplingRules();
+    TraceSamplingRules traceSamplingRules = TraceSamplingRules.EMPTY;
+    if (traceSamplingRulesJson != null) {
+      traceSamplingRules = TraceSamplingRules.deserialize(traceSamplingRulesJson);
+    } else {
+      traceSamplingRulesJson = "[]";
+    }
```

Comment:
Nitpick: I find it confusing the two branches did not update the same variable 😓 
 
```suggestion
    String traceSamplingRulesJson = config.getTraceSamplingRules();
    TraceSamplingRules traceSamplingRules;
    if (traceSamplingRulesJson == null) {
      traceSamplingRulesJson = "[]";
      traceSamplingRules = TraceSamplingRules.EMPTY;
    } else {
      traceSamplingRules = TraceSamplingRules.deserialize(traceSamplingRulesJson);
    }
```